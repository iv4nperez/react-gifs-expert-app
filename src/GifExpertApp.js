import React,{ useState } from 'react'
import { AddCategory } from './components/AddCategory'
import { GifGrid } from './components/GifGrid'

export const GifExpertApp = ({ defaulCategory = [] }) => {
    
    //const [ categories, setCategories ] = useState(['One Punch'])
    const [ categories, setCategories ] = useState( defaulCategory )
    
    return (
        <>
            <h2>GifExpertApp</h2>
            
            <AddCategory setCategories={ setCategories } />
            <br/>
            <hr/>
            <ol>
                {
                    categories.map( category => ( 
                        <GifGrid  
                            key={ category }    
                            category={ category } 
                        />
                    ))
                }
            </ol>
        </>
    )
}
