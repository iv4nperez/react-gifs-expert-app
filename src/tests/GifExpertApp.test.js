import React from 'react';
import  "@testing-library/jest-dom";
import { shallow } from "enzyme";
import { GifExpertApp } from '../GifExpertApp';

describe('Pruebas en GifExpertApp',() => {

    let wrapper = shallow( <GifExpertApp /> );
    test('debe de mopstrarse correctamente el componente', () => {

        expect( wrapper ).toMatchSnapshot();
    });
    
    test('debe de mostrar una lista de categorias', () => {
        
        const categories = ['One Punch','Dragon Ball']
        wrapper = shallow( <GifExpertApp defaulCategory={ categories } /> );
        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('GifGrid').length ).toBe( categories.length );
        
    });
    
});