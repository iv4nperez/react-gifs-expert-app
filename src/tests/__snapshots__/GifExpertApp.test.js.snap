// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifExpertApp debe de mopstrarse correctamente el componente 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <br />
  <hr />
  <ol />
</Fragment>
`;

exports[`Pruebas en GifExpertApp debe de mostrar una lista de categorias 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <br />
  <hr />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
  </ol>
</Fragment>
`;
